pkg/apiclient/offline-client.go

```diff
@@ -50,42 +49,43 @@ func newOfflineClient(paths []string) (context.Context, Client, error) {
 
 	for _, basePath := range paths {
 		err := file.WalkManifests(basePath, func(path string, bytes []byte) error {
-			var generic map[string]interface{}
-			if err := yaml.Unmarshal(bytes, &generic); err != nil {
-				return fmt.Errorf("failed to parse YAML from file %s: %w", path, err)
-			}
-			switch generic["kind"] {
-			case "ClusterWorkflowTemplate":
-				cwftmpl := new(wfv1.ClusterWorkflowTemplate)
-				if err := yaml.Unmarshal(bytes, &cwftmpl); err != nil {
-					return fmt.Errorf("failed to unmarshal file %s as a ClusterWorkflowTemplate: %w", path, err)
+			for _, pr := range common.ParseObjects(bytes, false) {
```

Comment:
I see this is copied from `lint.go`, so similarly I'd say this is fine for now, but a more clear name would be better in both places, like `parsed` instead of `pr` (which makes one think of "pull request")
```suggestion
			for _, parsed := range common.ParseObjects(bytes, false) {
```